lighthouse-core/report/html/renderer/element-screenshot-renderer.js

```diff
@@ -0,0 +1,240 @@
+      0, screenshotSize.width - elementPreviewSizeInScreenshotCoords.width
+    );
+    const screenshotTopVisisbleEdge = clamp(
+      elementRectCenter.y - elementPreviewSizeInScreenshotCoords.height / 2,
+      0, screenshotSize.height - elementPreviewSizeInScreenshotCoords.height
+    );
+
+    return {
+      screenshot: {
+        left: screenshotLeftVisibleEdge,
+        top: screenshotTopVisisbleEdge,
+      },
+      clip: {
+        left: elementRectInScreenshotCoords.left - screenshotLeftVisibleEdge,
+        top: elementRectInScreenshotCoords.top - screenshotTopVisisbleEdge,
+      },
+    };
+  }
+
+  /**
+   * @param {DOM} dom
+   * @param {HTMLElement} mask
+   * @param {{left: number, top: number}} positionClip
+   * @param {LH.Artifacts.Rect} elementRectInScreenshotCoords
+   * @param {Size} elementPreviewSizeInScreenshotCoords
+   */
+  static renderClipPath(dom, mask, positionClip,
+      elementRectInScreenshotCoords, elementPreviewSizeInScreenshotCoords) {
+    // Normalize values between 0-1.
+    const top = positionClip.top / elementPreviewSizeInScreenshotCoords.height;
+    const bottom =
+      top + elementRectInScreenshotCoords.height / elementPreviewSizeInScreenshotCoords.height;
+    const left = positionClip.left / elementPreviewSizeInScreenshotCoords.width;
+    const right =
+      left + elementRectInScreenshotCoords.width / elementPreviewSizeInScreenshotCoords.width;
+
+    const clipId = `clip-${top}-${bottom}-${left}-${right}`;
+    const clipPathSvg = dom.createElement('div');
+    clipPathSvg.innerHTML = `<svg height="0" width="0">
+        <defs>
+          <clipPath id='${clipId}' clipPathUnits='objectBoundingBox'>
+            <polygon points="0,0  1,0  1,${top} 0,${top}" ></polygon>
+            <polygon points="0,${bottom} 1,${bottom} 1,1 0,1" ></polygon>
+            <polygon points="0,${top} ${left},${top} ${left},${bottom} 0,${bottom}" ></polygon>
+            <polygon points="${right},${top} 1,${top} 1,${bottom} ${right},${bottom}" ></polygon>
+          </clipPath>
+        </defs>
+      </svg>`;
+
+    mask.style.clipPath = 'url(#' + clipId + ')';
+    mask.appendChild(clipPathSvg);
+  }
+
+  /**
+   * @param {DOM} dom
+   * @param {LH.Artifacts.FullPageScreenshot} fullPageScreenshot
+   */
+  static _installBackgroundImageStyle(dom, fullPageScreenshot) {
+    const containerEl = dom.find('.lh-container', dom.document());
+    if (containerEl.querySelector('#full-page-screenshot-style')) return;
+
+    const fullpageScreenshotUrl = fullPageScreenshot.data;
+    const fullPageScreenshotStyleTag = dom.createElement('style');
+    fullPageScreenshotStyleTag.id = 'full-page-screenshot-style';
+    fullPageScreenshotStyleTag.innerText = `
+      .lh-element-screenshot__image {
+        background-image: url(${fullpageScreenshotUrl})
+      }`;
+    containerEl.appendChild(fullPageScreenshotStyleTag);
+  }
+
+  /**
+   * @param {DOM} dom
```

Comment:
```suggestion
   * Installs the lightbox elements and wires up click listeners to all .lh-element-screenshot elements.
   * Should only be called exactly once per report render.
   * @param {DOM} dom
```

should we enforce the exactly once part?